server_installation/topics/operating-mode/crossdc/proc-setting-up-infinispan.adoc

```diff
@@ -0,0 +1,160 @@
+<3> Sets 1000 as the number of relay nodes for the cluster. You should set a value that is equal to or greater than the maximum number of nodes in your {jdgserver_name} cluster.
+<4> Names all {jdgserver_name} clusters that backup caches with {jdgserver_name} data and uses the default TCP stack for inter-cluster transport.
++
+.. Configure the {jdgserver_name} cluster transport to use the stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<cache-container name="default" statistics="true">
+      <transport cluster="${infinispan.cluster.name:cluster}"
+                 stack="xsite"/> <1>
+</cache-container>
+----
+<1> Uses the `xsite` stack for the cluster.
++
+. Configure the keystore as an SSL identity in the server security realm.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<server-identities>
+  <ssl>
+    <keystore path="server.jks" <1>
+              relative-to="infinispan.server.config.path"
+              keystore-password="password" <2>
+              alias="server" /> <3>
+  </ssl>
+</server-identities>
+----
+<1> Specifies the path of the keystore that contains the SSL identity.
+<2> Specifies the password to access the keystore.
+<3> Names the alias of the certificate in the keystore.
++
+. Configure the authentication mechanism for the Hot Rod endpoint.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<endpoints socket-binding="default" security-realm="default">
+   <hotrod-connector name="hotrod">
+      <authentication security-realm="default">
+         <sasl mechanisms="DIGEST-MD5 GSSAPI EXTERNAL" <1>
+               server-name="infinispan" /> <2>
+      </authentication>
+   </hotrod-connector>
+   <rest-connector name="rest"/>
+</endpoints>
+----
+<1> Configures the SASL authentication mechanism for the Hot Rod endpoint. You should use an appropriate mechanism for your environment. DIGEST-MD5 only is suitable for test and development environments.
+<2> Defines the name that {jdgserver_name} servers present to clients. You specify this name in the Hot Rod client configuration when you set up {jdgserver_name}.
++
+. Create a cache template.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<cache-container ... >
+  <replicated-cache-configuration name="sessions-cfg" <1>
+                                  mode="SYNC"> <2>
+    <locking acquire-timeout="0" /> <3>
+    <backups>
+      <backup site="NYC" strategy="ASYNC" /> <4>
+    </backups>
+  </replicated-cache-configuration>
+</cache-container>
+----
+<1> Creates a cache template named `sessions-cfg`.
+<2> Defines a cache that synchronously replicates data across the cluster.
+<3> Disables timeout for lock acquisition.
+<4> Names the backup site for the {jdgserver_name} cluster you are configuring.
++
+. Repeat the preceding steps to modify `infinispan.xml` on each node in the {jdgserver_name} cluster.
+. Start the cluster and open {jdgserver_name} Console in any browser.
```

Comment:
wouldn't be simpler to set up all these caches in `infinispan.xml` instead of making the user start the server and accessing the console?

Reply:
Interesting. This is something, which we discuss with Don in the JIRA. I've proposed to stick with infinispan.xml, but Don mentioned that it is not the best practice anymore. See the comments https://issues.redhat.com/browse/KEYCLOAK-16628?focusedCommentId=15646474&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15646474 and https://issues.redhat.com/browse/KEYCLOAK-16628?focusedCommentId=15647243&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15647243 .

My vote would be to stick with `infinispan.xml` as that is also what we use in our testsuite AFAIK. @tkyjovsk can confirm. And it would be ideal if the instructions here are exactly the one, which we test in our testsuite.

Reply:
@mposolda There was a little more discussion about this on the DG team. While it is possible to define caches in infinispan.xml that does have some drawbacks, notably if you want to add another cache in the xml, or modify a cache, then you have to bring the nodes down and modify the xml. 

I've been trying to advocate one consistent way to create caches with DG 8.x. It's sort of a "best practice" to create caches at runtime. See what's in the Infinispan Server guide here: https://infinispan.org/docs/stable/titles/server/server.html#create_remote_cache 

It's my opinion that we should have one recommended way to do things. I don't want to add a lot of friction to this issue, though, and I guess we should document what QE test. Please let us know what you think and I can work with Andy to modify the instructions as necessary. Thank you!

Reply:
I would go with Don's suggestion since it is the way to go for the openshift environment. For other environment, it may be "annoying" to do but it works as well.

Reply:
Okay, I will leave it as is unless I hear otherwise.